dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CensorHelper.java

```diff
@@ -0,0 +1,210 @@
+ * non-key data) will contain longer runs of the same character classes, and/or statistically
+ * resemble English text.
+ *
+ * The system is not perfect, but it does pretty well.  Experiments show that it fails to censor
+ * less than one in 50,000,000 randomly-generated 40-character strings composed of non-padding
+ * base64 characters.  It does censor the occasional non-secret thing, but for the current use
+ * case, that's ok.
+ *
+ * The advantage of this scheme over something like git secrets is that it censors secrets that
+ * are floating by themselves, having previously been separated from adjacent identifying
+ * markings or other cues.
+ */
+
+public class CensorHelper {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CensorHelper.class);
+    private static final Pattern BASE64_PATTERN = Pattern.compile("[-_a-zA-Z0-9+/]{22,}");
+    private static final Pattern HEX_PATTERN = Pattern.compile("^[a-fA-F0-9]*$");
+    private static final int LETTER_COUNT = 26;
+    private static final int LETTER_COUNT_SQUARED = LETTER_COUNT * LETTER_COUNT;
+    private static final int LETTER_COUNT_CUBED = LETTER_COUNT * LETTER_COUNT * LETTER_COUNT;
+
+    private final byte[] englishPoints;
+
+    public CensorHelper(Map<String, Double> tripletToFrequency) {
+        englishPoints = createPointArray(tripletToFrequency);
+    }
+
+    public String censor(String raw) {
+
+        StringBuilder censored = new StringBuilder(raw);
+
+        // Censor each longer base64-ish string if it's either all hex or is a "jumble" of characters.
+        // See the class description for more information about what we consider a "jumble".
+        Matcher matcher = BASE64_PATTERN.matcher(raw);
+        while (matcher.find()) {
+
+            int start = matcher.start();
+            int end = matcher.end();
+            String suspect = raw.substring(start, end);
+
+            if (HEX_PATTERN.matcher(suspect).matches() || isJumbled(suspect)) {
+                for (int i = start; i < end; i++) {
+                    censored.setCharAt(i, 'X');
+                }
+            }
+        }
+
+        return censored.toString();
+    }
+
+    private boolean isJumbled(String s) {
+        double length = s.length();
+        double adjacentsPerChar = calculateAdjacents(s) / length;
+        double englishPointsPerChar = calculateEnglishPoints(s) / length;
+        double score = adjacentsPerChar * adjacentsPerChar * (1 + 2 * englishPointsPerChar);
+        return score < 1.0;
+    }
+
+    private int calculateAdjacents(String s) {
+        int adjacents = 0;
+        for (int i = 0, n = s.length() - 1; i < n; i++) {
+            char a = s.charAt(i);
+            char b = s.charAt(i + 1);
+            if (characterClass(a) == characterClass(b)) {
+                adjacents++;
+            }
+        }
+        return adjacents;
+    }
+
+    private int characterClass(char c) {
+        if (c >= '0' && c <= '9') {
```

Comment:
These comparisons are hardcoded here because it's super-critical that these methods do exactly what we want them to, otherwise the code will crash or malfunction.  We can use the classes/methods you suggest, but we'll be a little less sure that the delegated operations are what we want to happen.

Reply:
Not sure I understand this, apache libraries are open source 
https://github.com/apache/commons-lang/blob/rel/commons-lang-3.14.0/src/main/java/org/apache/commons/lang3/CharSet.java#L72